src/components/pie-chart/pie-chart.js

```diff
@@ -1,32 +1,67 @@
 import template from "./pie-chart.html";
 import Chart from "chart.js";
 
+import { CurrencyConverterUAH } from "../../services/currency-converter-uah";
+import { ColorGenerator } from "../../services/color-generator";
+
 export class PieChartComponent {
   constructor(mountPoint) {
     this.mountPoint = mountPoint;
   }
 
   querySelectors() {
-    this.pieChartCtx = this.mountPoint.querySelector(".chart__visual");
+    this.chartCtx = this.mountPoint.querySelector(".chart__visual");
   }
 
-  mount() {
-    this.mountPoint.innerHTML = template();
-    this.querySelectors();
-    this.createPieChart();
+  update(data) {
+    if (data.type === "+") {
+      return;
+    }
+    this.dataset = this.addCurrData(this.dataset, data);
+    this.drawChanged();
   }
 
-  createPieChart() {
-    let pieChart = new Chart(this.pieChartCtx, {
+  createFromList(list) {
+    let expenceList = list.filter(item => item.type === "-").reverse();
+    this.dataset = expenceList.reduce(this.addCurrData, this.dataset);
+    this.drawChanged();
+  }
+
+  addCurrData(accum, item) {
+    let data = accum;
```

Comment:
if you just assign `accum` to `data` variable -- object stays the same (because you assigning object reference) -- so you're mutating params at the bottom -- please fix this